SU2_CFD/src/drivers/CDiscAdjMultizoneDriver.cpp

```diff
@@ -700,58 +701,103 @@ void CDiscAdjMultizoneDriver::DirectIteration(unsigned short iZone, RECORDING ki
                                            solver_container, numerics_container, config_container,
                                            surface_movement, grid_movement, FFDBox, iZone, INST_0);
 
+}
+
+void CDiscAdjMultizoneDriver::Print_DirectResidual(RECORDING kind_recording) {
+
+  /*--- Helper lambda func to return lenghty [iVar][iZone] string.  ---*/
+  auto iVar_iZone2string = [](unsigned short ivar, unsigned short izone) {
+    return "[" + std::to_string(ivar) + "][" + std::to_string(izone) + "]";
+  };
+
   /*--- Print residuals in the first iteration ---*/
 
   if (rank == MASTER_NODE && kind_recording == RECORDING::SOLUTION_VARIABLES) {
 
-    auto solvers = solver_container[iZone][INST_0][MESH_0];
+    const unsigned short fieldWidth = 15;
+    PrintingToolbox::CTablePrinter RMSTable(&std::cout);
 
-    switch (config_container[iZone]->GetKind_Solver()) {
+    /*--- The CTablePrinter requires two sweeps:
+     *--- 0. Add the colum names (addVals=0=false) plus CTablePrinter.PrintHeader()
+     *--- 1. Add the RMS-residual values (addVals=1=true) plus CTablePrinter.PrintFooter() ---*/
+    for (int addVals = 0; addVals < 2; addVals++) {
 
-      case DISC_ADJ_EULER:     case DISC_ADJ_NAVIER_STOKES:
-      case DISC_ADJ_INC_EULER: case DISC_ADJ_INC_NAVIER_STOKES:
-        cout << " Zone " << iZone << " (flow)       - log10[U(0)]    : "
-             << log10(solvers[FLOW_SOL]->GetRes_RMS(0)) << endl;
-        if (config_container[iZone]->AddRadiation()) {
+      for (unsigned short iZone = 0; iZone < nZone; iZone++) {
 
-          cout << " Zone " << iZone << " (radiation)  - log10[Rad(0)]  : "
-               << log10(solvers[RAD_SOL]->GetRes_RMS(0)) << endl;
-        }
-        break;
+        auto solvers = solver_container[iZone][INST_0][MESH_0];
+        auto configs = config_container[iZone];
```

Comment:
I am not sure whether one can easily swap those containers for singlezone and Multizone on this level

Reply:
Well I guess this works out as is ...